src/pages/filter/ui/query-builder/groups/chart/chart.store.ts

```diff
@@ -153,6 +137,28 @@ class ChartStore {
       return this.getVariantsChartData(subGroupItem)
     }
   }
+
+  getVariantListForPieChart(
+    filteredVariants: [string, number][],
+  ): [string, number][] {
+    const displayedFieldsNumber = 4
+    if (filteredVariants.length <= displayedFieldsNumber) {
+      return filteredVariants
+    }
+
+    const mainVariantList = filteredVariants.slice(0, displayedFieldsNumber)
+    const othersVariantList = filteredVariants.slice(displayedFieldsNumber)
+
+    const othersVariantsQuantity = othersVariantList.reduce(
+      (previousValue, variant) => previousValue + variant[1],
+      0,
+    )
+
+    const othersVaraints: [string, number] = ['Others', othersVariantsQuantity]
+    const dataForPieChart = [...mainVariantList, othersVaraints]
+
+    return dataForPieChart
+  }
 }
 
 export default new ChartStore()
```

Comment:
Why you use observable store for utility functions? It does not contain any data or computable values.

Reply:
Moved to utils